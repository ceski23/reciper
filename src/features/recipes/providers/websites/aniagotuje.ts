import aniagotujeIcon from 'assets/images/providers/ania_gotuje.png'
import { extractMicrodata } from 'features/recipes/providers/generic/microdata'
import type { RecipesProvider } from 'features/recipes/providers/scrapper'
import type { Recipe } from 'features/recipes/types'
import { isDefined } from 'lib/utils'
import { getTextFromNode, isElementNode } from 'lib/utils/dom'
import { getColorFromImage } from 'lib/utils/images'

export const aniagotuje: RecipesProvider = {
	name: 'AniaGotuje.pl',
	icon: aniagotujeIcon,
	matcher: /https:\/\/aniagotuje\.pl\/.*/,
	scrape: async doc => {
		const partialRecipe = await extractMicrodata(doc)
		const color = partialRecipe.image ? await getColorFromImage(partialRecipe.image) : undefined
		const name = doc.querySelector('.article-content [itemprop="name"]')?.textContent?.trim()

		return {
			...partialRecipe,
			color,
			name,
			ingredients: parseIngredients(doc) ?? partialRecipe.ingredients,
			instructions: parseInstructions(doc) ?? partialRecipe.instructions,
			gallery: parseImages(doc),
		}
	},
}

const parseImages = (doc: Document) =>
	Array
		.from(doc.querySelectorAll('[itemprop="recipeInstructions"] img'))
		.map(elem => elem.getAttribute('data-src'))
		.filter(isDefined)

const parseIngredients = (doc: Document) => {
	const root = doc.querySelector('[itemprop="recipeInstructions"] > .ads-slot-article + div')
	const ingredientsHeaders = root?.querySelectorAll<HTMLParagraphElement>('.ing-header')

	// Single recipe
	if (ingredientsHeaders?.length === 0) {
		const recipeIngredients = Array
			.from(root?.querySelectorAll('.recipe-ing-list [itemprop="recipeIngredient"]') ?? [])
			.map(elem => elem.textContent?.trim())
			.filter(isDefined)
			.map(text => ({ text }))

		return recipeIngredients.length > 0 ? recipeIngredients : undefined
	}

	// Recipe with sub-recipes
	const recipeIngredients = Array.from(ingredientsHeaders ?? []).reduce<{
		ingredients: Recipe['ingredients']
		group?: string
	}>(({ ingredients, group }, element) => {
		const nextGroup = element.textContent?.trim() ?? group
		const nextIngredients = Array
			.from(element.nextElementSibling?.querySelectorAll('[itemprop="recipeIngredient"]') ?? [])
			.map(elem => elem.textContent?.trim())
			.filter(isDefined)
			.map(text => ({ text, group: nextGroup }))

		return {
			ingredients: ingredients.concat(nextIngredients),
			group: nextGroup,
		}
	}, {
		ingredients: [],
		group: undefined,
	}).ingredients

	return recipeIngredients.length > 0 ? recipeIngredients : undefined
}

const parseInstructions = (doc: Document) => {
	const instructions: Recipe['instructions'] = []
	let currentElement: Node | Element | null = doc.querySelector('[itemprop="recipeInstructions"] div + :is(h3, h2)')
	let header: string | undefined

	while (currentElement !== null && currentElement !== undefined) {
		if (
			isElementNode(currentElement)
			&& ['H1', 'H2', 'H3', 'H4'].includes(currentElement.tagName)
		) {
			header = currentElement.textContent?.trim()
		} else {
			const text = getTextFromNode(currentElement)

			if (text !== null && text.length > 0) {
				instructions.push({ text, group: header })
			}
		}

		currentElement = currentElement.nextSibling
	}

	return instructions.length > 0 ? instructions : undefined
}
