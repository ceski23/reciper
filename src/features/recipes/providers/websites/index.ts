export { aniagotuje } from './aniagotuje'
export { kwestiasmaku } from './kwestiasmaku'
export { wszystkiegoslodkiego } from './wszystkiegoslodkiego'
