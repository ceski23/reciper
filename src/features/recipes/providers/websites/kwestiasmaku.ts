import kwestiasmakuIcon from 'assets/images/providers/kwestia_smaku.png'
import { extractMicrodata } from 'features/recipes/providers/generic/microdata'
import type { RecipesProvider } from 'features/recipes/providers/scrapper'
import type { Recipe } from 'features/recipes/types'
import { isDefined } from 'lib/utils'
import { getTextFromNode } from 'lib/utils/dom'
import { getColorFromImage } from 'lib/utils/images'
import { parseValidNumber } from 'lib/utils/numbers'

export const kwestiasmaku: RecipesProvider = {
	name: 'Kwestia Smaku',
	icon: kwestiasmakuIcon,
	matcher: /https:\/\/www\.kwestiasmaku\.com\/.*/,
	scrape: async doc => {
		const partialRecipe = await extractMicrodata(doc)
		const color = partialRecipe?.image ? await getColorFromImage(partialRecipe.image).catch(() => undefined) : undefined
		const description = doc.querySelector('.field-name-field-uwagi-wstepne')?.textContent?.trim()

		const tags = Array
			.from(doc.querySelectorAll('#node-przepis-full-group-kategorie a'))
			.map(elem => elem.textContent?.toLowerCase())
			.filter(isDefined)

		const servingsElement = doc.querySelector('.field-name-field-ilosc-porcji')
		const servings = parseValidNumber(servingsElement?.textContent)

		return {
			...partialRecipe,
			ingredients: parseIngredients(doc),
			instructions: parseInstructions(doc),
			color,
			description,
			tags,
			servings,
			gallery: parseImages(doc),
		}
	},
}

const parseIngredients = (doc: Document): Recipe['ingredients'] => {
	const ingredientsRoot = doc.querySelector('.field-name-field-skladniki')
	if (!ingredientsRoot) return []

	// Multiple ingredients groups
	if (ingredientsRoot.querySelector('.wyroznione')) {
		return Array.from(ingredientsRoot.children).reduce<{
			ingredients: Recipe['ingredients']
			group?: string
		}>(({ ingredients, group }, element) => {
			const nextGroup = (element.tagName === 'DIV' ? element.textContent?.trim() : undefined) ?? group
			const nextIngredients = Array
				.from(element.querySelectorAll('li'))
				.map(elem => getTextFromNode(elem).trim())
				.filter(isDefined)
				.map(text => ({ text, group: nextGroup }))

			return {
				ingredients: ingredients.concat(nextIngredients),
				group: nextGroup,
			}
		}, {
			ingredients: [],
			group: undefined,
		}).ingredients
	}

	// Signle ingredients group
	const recipeIngredients = Array
		.from(ingredientsRoot.querySelectorAll('ul > li'))
		.map(elem => elem.textContent?.trim())
		.filter(isDefined)
		.map(text => ({ text }))

	return recipeIngredients
}

const parseInstructions = (doc: Document): Recipe['instructions'] => {
	const instructionsRoot = doc.querySelector('.field-name-field-przygotowanie')
	if (!instructionsRoot) return []

	// Multiple instruction groups
	if (instructionsRoot.querySelector('.wyroznione')) {
		return Array.from(instructionsRoot.children).reduce<{
			instructions: Recipe['instructions']
			group?: string
		}>(({ instructions, group }, element) => {
			const nextGroup = (element.tagName === 'DIV' ? element.textContent?.trim() : undefined) ?? group
			const nextInstructions = Array
				.from(element.querySelectorAll('li'))
				.map(elem => getTextFromNode(elem).trim())
				.filter(isDefined)
				.map(text => ({ text, group: nextGroup }))

			return {
				instructions: instructions.concat(nextInstructions),
				group: nextGroup,
			}
		}, {
			instructions: [],
			group: undefined,
		}).instructions
	}

	// Signle instructions group
	return Array
		.from(instructionsRoot.querySelectorAll('ul > li'))
		.map(elem => getTextFromNode(elem).trim())
		.filter(isDefined)
		.map(text => ({ text }))
}

const parseImages = (doc: Document) =>
	Array
		.from(doc.querySelectorAll('.view-zdjecia img'))
		.map(elem => elem.getAttribute('src'))
		.filter(isDefined)
