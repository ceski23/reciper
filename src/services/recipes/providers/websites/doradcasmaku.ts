import icon from 'assets/providers/doradca_smaku.png';

import { Recipe } from 'services/recipes';
import type { Provider, RecipeScrapper } from 'services/recipes/providers';
import jsonldScrapper from 'services/recipes/providers/jsonld';

export const DoradcaSmakuProvider: Provider = (() => {
  const name = 'Doradca smaku';
  const url = 'https://www.doradcasmaku.pl/';

  const scrapper: RecipeScrapper = async (doc) => {
    const data = await jsonldScrapper(doc);

    const recipeImageElem = doc.querySelector('.recipe_item .recipe_gallery .img_wrapper img');
    const image = recipeImageElem?.getAttribute('data-src') ?? undefined;

    const partial: Partial<Recipe> = {
      image,
    };

    return Object.assign(data, partial);
  };

  return {
    name, url, scrapper, icon,
  };
})();
