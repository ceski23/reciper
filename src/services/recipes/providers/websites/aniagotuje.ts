import icon from 'assets/providers/ania_gotuje.png';

import { Recipe, RecipeIngredient, RecipeInstruction } from 'services/recipes';
import type { Provider, RecipeScrapper } from 'services/recipes/providers';
import microdataScrapper from 'services/recipes/providers/microdata';
import { colorExtractor } from 'services/recipes/providers/utils';

import { getTextFromNode } from 'utils/dom';
import { isElementNode, nonNullable } from 'utils/guards';
import { removeEmpty } from 'utils/objects';

export const AniaGotujeProvider: Provider = (() => {
  const name = 'AniaGotuje.pl';
  const url = 'https://aniagotuje.pl/';

  const parseIngredients = (doc: Document) => {
    const root = doc.querySelector('[itemprop="recipeInstructions"] > .ads-slot-article + div');
    const ingredientsHeaders = root?.querySelectorAll<HTMLParagraphElement>('.ing-header');
    const ingredients: RecipeIngredient[] = [];

    // Single recipe
    if (ingredientsHeaders?.length === 0) {
      const groupItemsElement = root?.querySelectorAll('.recipe-ing-list [itemprop="recipeIngredient"]');
      const groupItems = Array.from(groupItemsElement ?? [])
        .map((item) => item.textContent?.trim())
        .filter(nonNullable)
        .map<RecipeIngredient>((text) => ({ text }));

      ingredients.push(...groupItems);

      return ingredients;
    }

    // Recipe with sub-recipes
    ingredientsHeaders?.forEach((ingHeader) => {
      const groupTitle = ingHeader.textContent?.trim();

      const groupItemsElement = ingHeader.nextElementSibling?.querySelectorAll('[itemprop="recipeIngredient"]');
      const groupItems = Array.from(groupItemsElement ?? [])
        .map((item) => item.textContent?.trim())
        .filter(nonNullable)
        .map<RecipeIngredient>((text) => ({ text, group: groupTitle }));

      ingredients.push(...groupItems);
    });

    return ingredients;
  };

  const parseInstructions = (doc: Document) => {
    const instructions: RecipeInstruction[] = [];
    let currentElement: Node | Element | null = doc.querySelector('[itemprop="recipeInstructions"] .recipe-ing-list + h2');
    let header: string | undefined;

    while (currentElement !== null && currentElement !== undefined) {
      if (
        isElementNode(currentElement)
        && ['H1', 'H2', 'H3', 'H4'].includes(currentElement.tagName)
      ) {
        header = currentElement.textContent?.trim();
      } else {
        const text = getTextFromNode(currentElement);

        if (text !== null && text.length > 0) {
          instructions.push({ text, group: header });
        }
      }

      currentElement = currentElement.nextSibling;
    }

    return instructions;
  };

  // eslint-disable-next-line @typescript-eslint/no-shadow
  const scrapper: RecipeScrapper = async (doc) => {
    const data = await microdataScrapper(doc);

    let color;
    if (data?.image) {
      const palette = await colorExtractor(data.image);
      color = palette.Vibrant?.hex;
    }

    const recipeName = doc.querySelector('.article-content [itemprop="name"]')?.textContent?.trim();

    const partial: Partial<Recipe> = {
      color,
      name: recipeName,
      ingredients: parseIngredients(doc),
      instructions: parseInstructions(doc),
    };

    return Object.assign(data, removeEmpty(partial));
  };

  return {
    name, url, scrapper, icon,
  };
})();
