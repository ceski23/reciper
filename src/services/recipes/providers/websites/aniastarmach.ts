import icon from 'assets/providers/ania_starmach.png';

import type { Provider, RecipeScrapper } from 'services/recipes/providers';
import jsonldScrapper from 'services/recipes/providers/jsonld';
import { colorExtractor } from 'services/recipes/providers/utils';

export const AniaStarmachProvider: Provider = (() => {
  const name = 'Ania Starmach';
  const url = 'https://aniastarmach.pl/';

  // eslint-disable-next-line @typescript-eslint/no-shadow
  const scrapper: RecipeScrapper = async (doc) => {
    const data = await jsonldScrapper(doc);

    let color;
    if (data?.image) {
      const palette = await colorExtractor(data.image);
      color = palette.Vibrant?.hex;
    }

    const tagsElements = doc.querySelectorAll('.tags-wrapper a');
    const tags = Array
      .from(tagsElements)
      .map((elem) => elem.textContent?.toLowerCase())
      .filter(Boolean) as string[];

    return {
      ...data,
      color,
      tags,
    };
  };

  return {
    name, url, scrapper, icon,
  };
})();
