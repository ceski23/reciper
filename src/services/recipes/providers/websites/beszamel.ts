/* eslint-disable no-restricted-syntax */
import icon from 'assets/providers/beszamel.svg';

import { Recipe, RecipeIngredient } from 'services/recipes';
import type { Provider, RecipeScrapper } from 'services/recipes/providers';
import jsonldScrapper from 'services/recipes/providers/jsonld';
import { colorExtractor } from 'services/recipes/providers/utils';

import { getTextFromNode } from 'utils/dom';
import { nonNullable } from 'utils/guards';
import { removeEmpty } from 'utils/objects';

export const BeszamelProvider: Provider = (() => {
  const name = 'Beszamel';
  const url = 'https://beszamel.se.pl/';

  const parseIngredients = (doc: Document) => {
    const elements = doc.querySelectorAll('.ingredients__items li');
    let group: string | undefined;
    const ingredients: RecipeIngredient[] = [];

    for (const element of elements) {
      const text = getTextFromNode(element);

      if (text.endsWith(':')) {
        group = text;
      } else {
        ingredients.push(removeEmpty({ group, text }));
      }
    }

    return ingredients;
  };

  // eslint-disable-next-line @typescript-eslint/no-shadow
  const scrapper: RecipeScrapper = async (doc) => {
    const data = await jsonldScrapper(doc);

    let color;
    if (data?.image) {
      const palette = await colorExtractor(data.image);
      color = palette.Vibrant?.hex;
    }

    const tagsElements = doc.querySelectorAll('.breadcrumb .breadcrumb__item');
    const tags = Array
      .from(tagsElements)
      .map((elem) => elem.textContent?.toLowerCase().trim())
      .filter(nonNullable);

    const partial: Partial<Recipe> = {
      color,
      tags,
      ingredients: parseIngredients(doc),
    };

    return Object.assign(data, removeEmpty(partial));
  };

  return {
    name, url, scrapper, icon,
  };
})();
