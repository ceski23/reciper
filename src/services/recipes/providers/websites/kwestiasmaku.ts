/* eslint-disable @typescript-eslint/no-loop-func */
/* eslint-disable @typescript-eslint/brace-style */
/* eslint-disable no-restricted-syntax */
import icon from 'assets/providers/kwestia_smaku.png';

import { Recipe, RecipeIngredient, RecipeInstruction } from 'services/recipes';
import type { Provider, RecipeScrapper } from 'services/recipes/providers';
import microdataScrapper from 'services/recipes/providers/microdata';
import { getColorFromImage } from 'services/recipes/providers/utils';

import { getTextFromNode } from 'utils/dom';
import { nonNullable } from 'utils/guards';
import appLogger from 'utils/logger';
import { removeEmpty } from 'utils/objects';

const log = appLogger.extend('scrapper:kwestiasmaku');

export const KwestiaSmakuProvider: Provider = (() => {
  const name = 'Kwestia Smaku';
  const url = 'https://www.kwestiasmaku.com/';

  const parseIngredients = (doc: Document) => {
    const ingredientsRoot = doc.querySelector('.field-name-field-skladniki');
    if (!ingredientsRoot) return [];

    // Multiple ingredients groups
    if (ingredientsRoot.querySelector('.wyroznione')) {
      let group: string | undefined;
      const ingredients: RecipeIngredient[] = [];

      for (const element of ingredientsRoot.children) {
        if (element.tagName === 'DIV') {
          group = element.textContent?.trim() ?? undefined;
        } else {
          const recipeIngredients = Array
            .from(element.querySelectorAll('li'))
            .map((elem) => elem.textContent?.trim())
            .filter(nonNullable)
            .map<RecipeInstruction>((text) => ({ text, group }));

          ingredients.push(...recipeIngredients);
        }
      }

      return ingredients;
    }

    // Signle ingredients group
    const ingredientsElements = ingredientsRoot.querySelectorAll('ul > li');
    const recipeIngredients = Array
      .from(ingredientsElements)
      .map((elem) => elem.textContent?.trim())
      .filter(nonNullable)
      .map((text) => ({ text }));

    return recipeIngredients;
  };

  const parseInstructions = (doc: Document) => {
    const instructionsRoot = doc.querySelector('.field-name-field-przygotowanie');
    if (!instructionsRoot) return [];

    // Multiple instruction groups
    if (instructionsRoot.querySelector('.wyroznione')) {
      let group: string | undefined;
      const instructions: RecipeInstruction[] = [];

      for (const element of instructionsRoot.children) {
        if (element.tagName === 'DIV') {
          group = element.textContent?.trim() ?? undefined;
        } else {
          const recipeInstructions = Array
            .from(element.querySelectorAll('li'))
            .map((elem) => getTextFromNode(elem).trim())
            .filter(nonNullable)
            .map<RecipeInstruction>((text) => ({ text, group }));

          instructions.push(...recipeInstructions);
        }
      }

      return instructions;
    }

    // Signle instructions group
    const instructionsElements = instructionsRoot.querySelectorAll('ul > li');
    const recipeInstructions = Array
      .from(instructionsElements)
      .map((elem) => getTextFromNode(elem).trim())
      .filter(nonNullable)
      .map((text) => ({ text }));

    return recipeInstructions;
  };

  const parseImages = (doc: Document) => {
    log('looking for images');

    const imageElements = doc.querySelectorAll('.view-zdjecia img');
    const images = Array
      .from(imageElements)
      .map((elem) => elem.getAttribute('src'))
      .filter(nonNullable);

    log('found images:', images);

    return images;
  };

  // eslint-disable-next-line @typescript-eslint/no-shadow
  const scrapper: RecipeScrapper = async (doc) => {
    const microdata = await microdataScrapper(doc);
    const description = doc.querySelector('.field-name-field-uwagi-wstepne')?.textContent?.trim() || undefined;

    const tagsElements = doc.querySelectorAll('#node-przepis-full-group-kategorie a');
    const tags = Array
      .from(tagsElements)
      .map((elem) => elem.textContent?.toLowerCase())
      .filter(nonNullable);

    const servingsElement = doc.querySelector('.field-name-field-ilosc-porcji');
    const servings = servingsElement?.textContent
      ? Number.parseInt(servingsElement.textContent, 10)
      : undefined;

    const color = microdata.image ? await getColorFromImage(microdata.image) : undefined;

    const partial: Partial<Recipe> = {
      ingredients: parseIngredients(doc),
      instructions: parseInstructions(doc),
      color,
      description,
      tags,
      servings,
      gallery: parseImages(doc),
    };

    return Object.assign(microdata, removeEmpty(partial));
  };

  return {
    name, url, scrapper, icon,
  };
})();
