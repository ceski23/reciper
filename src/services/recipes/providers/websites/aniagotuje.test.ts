import { expect } from 'vitest';

import { tortTruskawkowy } from 'services/recipes/providers/fixtures/aniagotuje';
import { AniaGotujeProvider } from 'services/recipes/providers/websites/aniagotuje';

const scrapeRecipe = async (data: string) => {
  const parser = new DOMParser();
  const doc = parser.parseFromString(data, 'text/html');
  const partialRecipe = await AniaGotujeProvider.scrapper(doc);
  partialRecipe.id = '1';

  return partialRecipe;
};

describe('scrapper', () => {
  it('should scrape valid recipe with ingredients groups', async () => {
    const recipe = await scrapeRecipe(tortTruskawkowy);
    expect(recipe).toMatchSnapshot();
  });
});
