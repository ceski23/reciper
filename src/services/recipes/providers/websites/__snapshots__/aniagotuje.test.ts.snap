// Vitest Snapshot v1

exports[`scrapper > should scrape valid recipe with ingredients groups 1`] = `
{
  "calories": 370,
  "color": "#7f7f7f",
  "description": "Tort truskawkowy na jasnym biszkopcie, z pysznym i super prostym do zrobienia kremem śmietanowym z mascarpone to rewelacyjny pomysł na deser oraz na urodzinowy tort. Ten przepis sprawdzi się przez cały rok.",
  "id": "1",
  "image": "https://cdn.aniagotuje.com/pictures/articles/2022/06/28889245-v-1500x1500.jpg",
  "ingredients": [
    {
      "group": "Składniki na biszkopt",
      "text": "niepełna szklanka mąki pszennej tortowej - około 140 g",
    },
    {
      "group": "Składniki na biszkopt",
      "text": "1/3 szklanki mąki ziemniaczanej - do 50 g skrobi",
    },
    {
      "group": "Składniki na biszkopt",
      "text": "7 większych jajek - do 400 g po rozbiciu",
    },
    {
      "group": "Składniki na biszkopt",
      "text": "3/4 szklanki drobnego cukru - około 150 g",
    },
    {
      "group": "Składniki na krem z mascarpone",
      "text": "750 g serka mascarpone",
    },
    {
      "group": "Składniki na krem z mascarpone",
      "text": "600 ml śmietanki kremówki 36%",
    },
    {
      "group": "Składniki na krem z mascarpone",
      "text": "120 g cukru pudru",
    },
    {
      "group": "Pozostałe składniki",
      "text": "700 g truskawek - świeże lub mrożone",
    },
    {
      "group": "Pozostałe składniki",
      "text": "2 łyżki drobnego cukru - do 35 g",
    },
    {
      "group": "Pozostałe składniki",
      "text": "3 łyżki soku z cytryny",
    },
    {
      "group": "Do dekoracji",
      "text": "świeże truskawki; listki mięty; pokruszone mini bezy - u mnie z orzechami",
    },
  ],
  "instructions": [
    {
      "group": "Tort truskawkowy",
      "text": "Szklanka ma u mnie pojemność 250 ml.
Jajka wyjmij z lodówki na godzinę przed szykowaniem biszkoptu.
Serek mascarpone oraz śmietanka kremówka 36 % powinny być trzymane w lodówce. Oba te składniki muszą być mocno schłodzone.",
    },
    {
      "group": "Tort truskawkowy",
      "text": "Forma do pieczenia: tortownica o średnicy 26 cm.
Pieczenie: 180 stopni, środkowa półka z opcją pieczenia góra/dół
Czas pieczenia: około 30 minut.",
    },
    {
      "group": "Tort truskawkowy",
      "text": "Kalorie policzone zostały na podstawie użytych przeze mnie składników. Jest to więc orientacyjna liczba kalorii, ponieważ Twoje składniki mogą mieć inną liczbę kalorii niż te, których użyłam ja. Z podanej ilości składników otrzymasz tort truskawkowy o wadze minimum 2000 gramów. Zanim zaczniesz szykować przepis na tort z truskawkami, polecam najpierw przeczytać cały przepis, by przygotować się odpowiednio wcześniej do kolejnych etapów wykonania. Można bowiem upiec wcześniej biszkopt lub też z wyprzedzeniem przygotować sobie krem i trzymać go w lodówce.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "W pierwszej kolejności upiecz biszkopt. Możesz to zrobić nawet na dwa dni przed szykowaniem całego tortu truskawkowego. Jeśli potrzebujesz super szczegółowej instrukcji, to zapraszam tu - przepis na biszkopt , gdzie podaję bardzo dokładny przepis na taki sam biszkopt. Poniżej zaś skrócona instrukcja wykonania.

Aby zrobić biszkopt, w osobnej misce wymieszaj ze sobą obie mąki np. przy pomocy rózgi kuchennej. Można je też razem przesiać przez sitko.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Oddziel żółtka od białek. Białka umieść w czystym i suchym naczyniu i zacznij je ubijać przy pomocy miksera zaczynając od mniejszych obrotów. Stopniowo je zwiększaj. Białka ubijaj tak chwilę, by nie przebić pęcherzyków powietrza, a następnie zacznij dodawać cukier drobny. Po każdej łyżce miksuj białko na najwyższych obrotach, aż cukier połączy się z pianą i całkowicie się w niej rozpuści. Proces ubijania jest dość długi i może trwać łącznie nawet kilkanaście minut. Pod koniec ubijania powinna powstać gładka, gęsta i bardzo błyszcząca masa białkowa (zupełnie jak na bezę).",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Na koniec dodaj też odłożone do osobnej miski żółtka i dalej miksuj całość jeszcze chwilę, by białka gładko połączyły się z żółtkami. Obroty miksera zmniejsz do minimum (lub użyj szpatułki). Do piany jajecznej dodaj mieszankę mąk. Całość zamieszaj dokładnie, ale nie za długo, tylko do połączenia składników.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Dno tortownicy o średnicy 26 cm wystarczy wyłożyć papierem do pieczenia i zapiąć obręcz. Boków formy niczym nie smarujemy. Ciasto biszkoptowe przełóż do tortownicy i delikatnie wyrównaj jego powierzchnię. Biszkopt piecz około 30 minut (raczej nie dłużej niż 35 minut) w piekarniku nagrzanym do 170 stopni z termoobiegiem (do suchego patyczka). Z grzałką góra/dół piekarnik nagrzej wcześniej do 180 stopni. Biszkopt możesz wyjąć z piekarnika od razu po upieczeniu. Odczekaj przynajmniej 10 minut zanim wyjmiesz go z formy. Ostrożnie oddziel płaskim nożem brzegi ciasta od wewnętrznej części formy. Biszkopt odwróć do studzenia na drugą stronę. Jest ona równa i ładniej będzie wyglądała jako góra tortu. Ta górna zaś wyrówna się pod ciężarem ciasta.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Porada",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": ": Biszkopt ten można upiec nawet na trzy dni przed planowanym podaniem tortu. Przestudzony biszkopt zawiń luźno w papier śniadaniowy i trzymaj w torbie spożywczej w suchym i chłodnym miejscu.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Po całkowitym przestudzeniu biszkopt podziel na trzy blaty (placki z biszkoptu) np. specjalnym nożem strunowym do krojenia blatów. Na pierwszy blat (spód tortu) kładę smażone truskawki. Aby je zrobić w małym garnku, czy nawet na patelni, umieść 500 gramów umytych i odszypułkowanych truskawek. Warto je wcześniej drobno pokroić (kostka lub plasterki). Na patelni umieść też dwie łyżki cukru. Ustaw średnią moc palnika i zacznij smażyć truskawki, cały czas mieszając je np. przy pomocy drewnianej łyżki. W ciągu 10-15 minut całkowicie zmiękną. Pod koniec smażenia dodaj do nich trzy łyżki świeżo wyciśniętego soku z cytryny. Zamieszaj całość i wyłącz.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Porada",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": ": Można też sięgnąć po truskawki mrożone. Zalecam rozmrozić je wcześniej i przed smażeniem odlać rozpuszczoną glazurę. W ostateczności do posmarowania spodu można też użyć dżemu truskawkowego. Wystarczy słoiczek o wadze około 350 gramów.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Gorące truskawki warto przelać do osobnego naczynia, by szybciej przestygły. Wyszło mi dokładnie 375 gramów smażonych truskawek (razem z sokiem). Dokładnie tak samo możesz postępować z malinami, jagodami oraz z obranymi ze skórki morelami, czy brzoskwiniami.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Dolny blat biszkoptu - spód, umieść na paterze, czy też na specjalnym jednorazowym \\"talerzu\\" papierowym do podawania tortów. Ponieważ na ten blat wykładasz smażone truskawki, które mają dużo soku/syropu, to tego blatu nie nasączasz ponczem. Przestudzone, gęste truskawki rozłóż w miarę równo po całym spodzie. Jeśli wydaje Ci się, że wyszło ich za dużo, to nadmiar możesz śmiało zostawić do udekorowania drugiego blatu.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Na warstwę smażonych truskawek wykładasz jeszcze krem śmietankowy z mascarpone. Aby go zrobić należy umieścić razem w jednym, wysokim naczyniu wszystkie składniki. Będzie to: 750 gramów dobrze schłodzonego serka mascarpone (np. jedno opakowanie 500 gramów + jedno opakowanie 250 gramów); 600 ml śmietanki kremówki 36%; 120 gramów cukru pudru - 12 płaskich łyżek. Rozpocznij ubijanie kremu. Najpierw na niższych obrotach miksera. Po chwili zwiększaj obroty, aż do maksymalnej mocy. Krem powinien być gotowy błyskawicznie. Masa jest dobrze ubita, gdy zaczyna stanowić opór na mieszadłach miksera oraz po nałożeniu na palec zachowuje strukturę - nie rozlewa się, nie przemieszcza. Przy tak tłustej śmietance z serkiem wystarczy nawet tylko minuta ubijania.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Porada : Na 20-30 minut przed miksowaniem, naczynie ze składnikami warto umieścić dodatkowo w komorze zamrażalnika. Krem można śmiało przygotować nawet dzień wcześniej i trzymać w lodówce pod przykryciem. Jeśli wolisz śmietankę 30 %, to możesz jej użyć. Taką śmietankę warto jednak najpierw ubić na sztywno z całym cukrem pudrem, a na koniec łyżka po łyżce wmiksować też mascarpone. Krem wyjdzie trochę mniej kaloryczny, a do tego bardziej puszysty.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Na warstwę smażonych truskawek zacznij wykładać około 1/4 całego kremu. Ostrożnie i delikatnie wyrównaj powierzchnię. Na krem wyłóż drugi blat biszkoptowy. Ten blat polecam już nasączyć ponczem. Do nasączenia tego tortu używam tylko przegotowanej wody wymieszanej z sokiem z cytryny (na 100 ml wody daję dwie łyżki soku z cytryny). Biszkopt nasącz z umiarem np. przy pomocy małej łyżeczki lub też przy użyciu specjalnej butelki z dziurkami.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Na drugim blacie rozsmaruj delikatnie kolejną warstwę kremu. Znowu około 1/4 kremu lub tym razem już trochę więcej. Druga porcję kremu możesz też wymieszać z kawałkami świeżych truskawek lub rozłożyć je na wierzchu w formie cienkich plasterków. Na kremie umieść trzeci, ostatni blat biszkoptowy. Ten blat również należy nasączyć.

Porada : Wody z cytryną na poncz nie podawałam już w wykazie składników. Możesz też nasączyć blaty biszkoptowe wodą z cukrem i alkoholem, czy nawet jasnym napojem gazowanym.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Trzeci blat przykryj warstwą kremu, zaś pozostały krem rozprowadź też na bokach tortu. Kremu jest bardzo dużo, zatem śmiało wystarczy Ci go do otynkowania całego tortu truskawkowego. Z reszty truskawek polecam wykonać dekorację. Truskawki pokrój w plasterki nie grubsze niż 4 mm (kroisz wzdłuż truskawki). Plasterki przykładaj do boków tortu przy samym dole i wciskaj lekko w krem, by powstał wzór jak na zdjęciu poniżej.",
    },
    {
      "group": "Tort truskawkowy przepis",
      "text": "Górę tortu udekorowałam kawałkami truskawek, listkami mięty oraz pokruszonymi bezami. Tym razem były to mini bezy pieczone z orzechami ziemnymi.",
    },
  ],
  "name": "Tort truskawkowy",
  "prepTime": 90,
  "rating": 4.9,
  "servings": 10,
  "tags": [
    "sylwester",
    "walentynki",
    "cukier",
    "cukier puder",
    "cytryna",
    "jajka",
    "mascarpone",
    "mąka pszenna tortowa",
    "skrobia ziemniaczana",
    "truskawki",
    "śmietanka",
  ],
}
`;
